src/mds/nameserver2/curvefs.h

```diff
@@ -803,6 +811,11 @@ class CurveFS {
     bool IsDefaultThrottleParams(const FileThrottleParams &params,
                                  uint64_t length) const;
 
+    bool IsLock(const OpenFileContext* cont) {
+        return nullptr != cont && cont->openflags()
+            == CurveOpenFlags::CURVE_RDWR;
+    }
```

Comment:
currently, I think we should only return true iff flags has RDWR and SHARED. And, it should be a static function.